File: app/src/test/java/kztproject/jp/splacounter/api/TodoistClientTest.java
package kztproject.jp.splacounter.api;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.runners.MockitoJUnitRunner;

import io.reactivex.Single;
import kztproject.jp.splacounter.DummyCreator;
import kztproject.jp.splacounter.model.UserResponse;

import static org.mockito.Mockito.anyString;
import static org.mockito.Mockito.when;

@RunWith(MockitoJUnitRunner.class)
public class TodoistClientTest {

    @Mock
    private TodoistService service;

    @InjectMocks
    private TodoistClient client;

    @Test
    public void getUserSuccess() {
        UserResponse dummyUserResponse = DummyCreator.createDummyUserResponse();
        when(service.getUser(anyString(), anyString(), anyString()))
                .thenReturn(Single.just(dummyUserResponse));

        client.getUser("test")
                .test()
                .assertNoErrors()
                .assertResult(dummyUserResponse)
                .assertComplete();
    }

}

File: app/src/test/java/kztproject/jp/splacounter/api/TodoistClientTest.kt
package kztproject.jp.splacounter.api

import com.nhaarman.mockito_kotlin.whenever
import io.reactivex.Single
import kztproject.jp.splacounter.DummyCreator
import org.junit.Before
import org.junit.Test
import org.mockito.ArgumentMatchers.anyString
import org.mockito.Mockito

class TodoistClientTest {

    private val service: TodoistService = Mockito.mock(TodoistService::class.java)

    private lateinit var client: TodoistClient

    @Before
    fun setup() {
        client = TodoistClient(service)
    }

    @Test
    fun getUserSuccess() {
        val dummyUserResponse = DummyCreator.createDummyUserResponse()
        whenever(service.getUser(anyString(), anyString(), anyString()))
                .thenReturn(Single.just(dummyUserResponse))

        client.getUser("test")
                .test()
                .assertNoErrors()
                .assertResult(dummyUserResponse)
                .assertComplete()
    }

}
